lib/posix/clock.c

```diff
@@ -26,6 +40,9 @@ int clock_gettime(clockid_t clock_id, struct timespec *ts)
 {
 	u64_t elapsed_msecs;
 	struct timespec base;
+#ifdef CONFIG_RTC
+	struct device *dev_rtc;
+#endif
 
 	switch (clock_id) {
 	case CLOCK_MONOTONIC:
```

Comment:
so RTC is used only for REALTIME. Could that be an issue in some cases? It means that those clock ids are using different source clock thus accuracy. I'm not sure what are the typical uses cases but can imagine that someone reads both at one point of time and next time in an hour. Monotonic will show different time elapsed than realtime.

Reply:
In my system I need to timestamp some data precisely.  My system could run during a long time without time synchronization. The RTC have a lower clock drift than system clock.

Reply:
exactly, thus i wonder if it is possible to provide monotonic based on RTC device. Maybe logging uptime offset when settime is called?

Reply:
There is 2 problems to use RTC as monotonic:
- According to Posix: "the MONOTONIC clock is not affected by discontinuous jumps in the system time". (A workaround will be to save the offset on settime and use it to correct monotonic clock)
- The resolution for shorter time is better with system clock